src/libs/OptionsListUtilsLHN.js

```diff
@@ -0,0 +1,312 @@
+            && !reportContainsIOUDebt) {
+            return false;
+        }
+
+        // We let Free Plan default rooms to be shown in the App - it's the one exception to the beta, otherwise do not show policy rooms in product
+        if (ReportUtils.isDefaultRoom(report) && !Permissions.canUseDefaultRooms(betas) && ReportUtils.getPolicyType(report, policies) !== CONST.POLICY.TYPE.FREE) {
+            return false;
+        }
+
+        if (ReportUtils.isUserCreatedPolicyRoom(report) && !Permissions.canUsePolicyRooms(betas)) {
+            return false;
+        }
+
+        if (isPolicyExpenseChat && !Permissions.canUsePolicyExpenseChat(betas)) {
+            return false;
+        }
+
+        return true;
+    });
+
+    let orderedReports = _.sortBy(filteredReports, sortByTimestampDescending ? 'lastMessageTimestamp' : 'reportName');
+
+    if (sortByTimestampDescending) {
+        orderedReports.reverse();
+    }
+
+    // Move the archived Rooms to the last
+    orderedReports = _.sortBy(orderedReports, report => ReportUtils.isArchivedRoom(report));
+
+    // Put all the reports into the different buckets
+    for (let i = 0; i < orderedReports.length; i++) {
+        const report = orderedReports[i];
+
+        // If the report is pinned and we are using the option to display pinned reports on top then we need to
+        // collect the pinned reports so we can sort them alphabetically once they are collected
+        if (report.isPinned) {
+            pinnedReportOptions.push(report);
+        } else if (report.hasOutstandingIOU && !report.isIOUReportOwner) {
+            iouDebtReportOptions.push(report);
+        } else if (report.hasDraft) {
+            draftReportOptions.push(report);
+        } else {
+            recentReportOptions.push(report);
+        }
+    }
+
+    // Prioritizing reports with draft comments, add them before the normal recent report options
+    // and sort them by report name.
+    const sortedDraftReports = _.sortBy(draftReportOptions, 'text');
+    recentReportOptions = sortedDraftReports.concat(recentReportOptions);
+
+    // Prioritizing IOUs the user owes, add them before the normal recent report options and reports
+    // with draft comments.
+    const sortedIOUReports = _.sortBy(iouDebtReportOptions, 'iouReportAmount').reverse();
+    recentReportOptions = sortedIOUReports.concat(recentReportOptions);
+
+    // If we are prioritizing our pinned reports then shift them to the front and sort them by report name
+    const sortedPinnedReports = _.sortBy(pinnedReportOptions, 'text');
+    recentReportOptions = sortedPinnedReports.concat(recentReportOptions);
+
+    return _.chain(recentReportOptions)
+        .pluck('reportID')
+        .map(reportID => reportID.toString())
+        .value();
+}
+
+/**
+ * Gets all the data necessary for rendering an OptionRowLHN component
+ *
+ * @param {String} reportID
+ * @returns {Object}
+ */
+function getOptionData(reportID) {
```

Comment:
similar question to the other, but it looks like this duplicates a lot of logic from `createOption()`?

Reply:
Yes, it does 👍 That's on purpose for now :D